packages/skeleton-react/src/lib/components/Rating/Rating.tsx

```diff
@@ -0,0 +1,196 @@
+import React, { createContext, useCallback, useContext, useEffect, useRef, useState } from "react";
+import { IconProps, RatingContextState, RatingProps } from "./types";
+
+// Context
+
+export const RatingContext = createContext<RatingContextState>({
+    interactive: false,
+    value: 0,
+    order: 0,
+    figureRef: undefined
+});
+
+// Components ---
+const RatingRoot: React.FC<RatingProps> = ({
+    value = 0,
+    max = 5,
+    interactive = false,
+    fraction = 1,
+    // Root
+    base = 'flex',
+    width = 'w-full',
+    text = 'text-token',
+    fill = 'fill-token',
+    justify = 'justify-center',
+    spaceX = 'space-x-2',
+    classes = '',
+    // Item ---
+    buttonBase = 'w-full h-full',
+    buttonPosition = 'relative',
+    buttonAspect = 'aspect-square',
+    buttonClasses = '',
+    // Events
+    onMouseDown = () => {},
+    onKeyDown = () => {},
+    onValueChange = () => {},
+	// Children
+	children,
+}) => {
+    const figureRef = useRef<HTMLElement>(null);
+
+    const [focusedButtonIndex, setFocusedButtonIndex] = useState(0);
```

Comment:
Keep your `useState` definitions grouped at the top, similar to `$state()` runes or vars in general.